services/spar/test-integration/Test/LoggingSpec.hs

```diff
@@ -18,3 +26,33 @@ spec = describe "logging" $ do
         Log.flush logger
       out `shouldContain` "hrgh  woaa"
       out `shouldNotContain` "hrgh\n\nwoaa"
+
+  let withApp :: (Application -> Env -> Expectation) -> TestSpar ()
+      withApp testcase = do
+        (app, env) <- liftIO . mkApp =<< asks (^. teOpts)
+        liftIO $ testcase app env
+
+      test400 :: HW.WaiSession HW.SResponse -> Int -> TestSpar ()
+      test400 badReq expectedStatus = withApp $ \app env -> do
```

Comment:
This test is extremely fragile in that if we change the `Expected exception instead of response with status >= 400` string by even a single character then this test will always pass no matter what even if it shouldn't. Kinda the opposite of a flaky test haha.

Reply:
`test400` isn't clear about what it does. Why does it accept a status-code if it's supposedly hard-coded to `400`? Even reading the function I'm not quite sure what it does 🤔